packages/auth-client/src/index.ts

```diff
@@ -284,6 +289,23 @@ export class AuthClient {
 
     // Open a new window with the IDP provider.
     this._idpWindow = window.open(identityProviderUrl.toString(), 'idpWindow') ?? undefined;
+
+    // Check if the _idpWindow is closed by user.
+    this._checkUserInterrupt(() => this._handleFailure(ERROR_USER_INTERRUPT, options?.onError));
+  }
+
+  private _checkUserInterrupt(onInterrupt: () => void) {
```

Comment:
Since this is used in only one case, I would probably avoid passing the `onInterrupt` callback here and consolidate the `_handleFailure` logic into `_checkUserInterrupt`. Passing onInterrupt to a private method is probably a premature abstraction.

Otherwise, this is a solid proposal, and will be a nice hook for UI's

Reply:
Thanks for reviewing. I've updated according to your comment.

One more test case is added.
I also add a line in the login success case
```typescript
it('should call onSuccess if recieved a valid success message', async () => {
 ...
 await client.logout();
});
```
otherwise, the following test cases would throw exception `Error: Invalid public key.`